libnmstate/schemas/interfaces.json

```diff
@@ -10,21 +10,89 @@
                 "required": ["name"]
             }
         }
     },
-    "required": ["interfaces"]
+    "required": ["interfaces"],
+
+    "definitions": {
+        "interface-rw": {
+            "properties": {
+                "name": { "type": "string" },
+                "description": { "type": "string" },
+                "state": {
+                    "type": "string",
+                    "enum": ["absent", "up", "down", "unknown"]
+                },
+                "link-speed": {
+                    "title": "Speed in Mbps",
+                    "type": "integer",
+                    "minimum": 0
+                },
+                "mac-address": { "$ref": "#/definitions/values/mac-address" },
+                "mtu": {
+                    "title": "Maximum transmit unit",
+                    "type": "integer",
+                    "minimum": 0
+                }
+            }
+        },
+        "interface-ro": {
+            "properties": {
+                "if-index": { "type": "integer", "minimum": 0, "readOnly": true },
+                "admin-status": { "allOf": [ { "$ref": "#/definitions/values/status" }, { "readOnly": true } ] },
+                "link-status": { "allOf": [ { "$ref": "#/definitions/values/status" }, { "readOnly": true } ] },
+                "phys-address": { "allOf": [ { "$ref": "#/definitions/values/mac-address" }, { "readOnly": true } ] },
+                "higher-layer-if": { "type": "string", "readOnly": true },
+                "lower-layer-if": { "type": "string", "readOnly": true },
+                "statistics": {
+                    "title": "Interface statistics",
+                    "type": "object",
+                    "properties": {
+                        "in-octets": { "type": "integer", "minimum": 0 },
+                        "in-unicast-pkts": { "type": "integer", "minimum": 0 },
+                        "in-broadcast-pkts": { "type": "integer", "minimum": 0 },
+                        "in-multicast-pkts": { "type": "integer", "minimum": 0 },
+                        "in-discards": { "type": "integer", "minimum": 0 },
+                        "in-errors": { "type": "integer", "minimum": 0 },
+                        "out-octets": { "type": "integer", "minimum": 0 },
+                        "out-unicast-pkts": { "type": "integer", "minimum": 0 },
+                        "out-broadcast-pkts": { "type": "integer", "minimum": 0 },
+                        "out-multicast-pkts": { "type": "integer", "minimum": 0 },
+                        "out-discards": { "type": "integer", "minimum": 0 },
+                        "out-errors": { "type": "integer", "minimum": 0 }
+                    },
+                    "readOnly": true,
+                    "required": ["in-octets", "in-unicast-pkts", "in-broadcast-pkts", "in-multicast-pkts", "in-discards", "in-errors", "out-octets", "out-unicast-pkts", "out-broadcast-pkts", "out-multicast-pkts", "out-discards", "out-errors"]
```

Comment:
could you please write each requirement on one line? so we can easily sort them?